velox/functions/sparksql/Decimal.cpp

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "velox/functions/sparksql/Decimal.h"
+
+#include "velox/expression/DecodedArgs.h"
+
+namespace facebook::velox::functions::sparksql {
+namespace {
+
+// Return the unscaled bigint value of a decimal, assuming it
+// fits in a bigint. Only short decimal input is accepted.
+class UnscaledValueFunction final : public exec::VectorFunction {
+  void apply(
+      const SelectivityVector& rows,
+      std::vector<VectorPtr>& args,
+      const TypePtr& outputType,
+      exec::EvalCtx& context,
+      VectorPtr& resultRef) const final {
+    VELOX_CHECK_EQ(args.size(), 1);
+    VELOX_CHECK(
+        args[0]->type()->isShortDecimal(), "ShortDecimal type is required.");
+    exec::DecodedArgs decodedArgs(rows, args, context);
+    auto decimalVector = decodedArgs.at(0);
+    context.ensureWritable(rows, BIGINT(), resultRef);
+    auto result =
+        resultRef->asUnchecked<FlatVector<int64_t>>()->mutableRawValues();
+    rows.applyToSelected(
+        [&](int row) { result[row] = decimalVector->valueAt<int64_t>(row); });
+  }
+};
+} // namespace
+
+std::vector<std::shared_ptr<exec::FunctionSignature>>
+unscaledValueSignatures() {
+  return {exec::FunctionSignatureBuilder()
+              .integerVariable("a_precision")
```

Comment:
Hmm... this signature allows both short and long decimals. CC: @majetideepak

Reply:
Can the constraint parameter of the integerVariable method be limited to only short decimal?

Reply:
Currently, the only option is to add short decimal validation within the function itself.